### DISCUSSION TITLE
Limit amount of posts

### DISCUSSION THREAD

**Discussion by timmohlin** [vercel/next.js#14939] [Help] [answered]
Hi,

Imagine that I would only like to show the latest two blog posts on the homepage, where would I most efficiently handle that?
In other words, where would I limit the data fetching to two posts rather than all posts in an easy and accurate way in NextJS?

I am using the NextJS starter-template (https://nextjs.org/learn/basics/data-fetching/implement-getstaticprops)

Any advice, best practice or docs on this?

Thanks!

**Comment by jamesmosier** [comment, marked as answer]
You could do this in `getStaticProps` for your `pages/index.js` file. 

In the example, the posts are sorted by date:

```js
  return allPostsData.sort((a, b) => {
    if (a.date < b.date) {
      return 1
    } else {
      return -1
    }
  })
```

and instead of returning all of them, just return the most recent 2 using `.slice(0,2)`:

```js
  return allPostsData.sort((a, b) => {
    if (a.date < b.date) {
      return 1
    } else {
      return -1
    }
  }).slice(0,2)
```
